### ISSUE TITLE
stop existing, possibly in favor of briefcase

### ISSUE THREAD

**Issue by glyph** [glyph/Encrust#2]
briefcase issues that would need to be resolved to kill Encrust completely:

- https://github.com/beeware/briefcase/issues/2180
  - this is needed to embed Sparkle, but perhaps Encrust could continue to exist as a thing that would automate running the thing to generate appcasts for me, since that is inherently platform-specific and fairly customized for me (Encrust doesn't have any code that would readily generalize to a more sophisticated release workflow yet)?
- https://github.com/glyph/Encrust/issues/10
  - I need to do my own investigating here to see if e.g. Pomodouroboros could exclude a few directories and get within throwing distance of its current bundle size

-----
## Already Completed
 
- https://github.com/beeware/briefcase/issues/1212
- https://github.com/beeware/briefcase/issues/1217
- https://github.com/beeware/briefcase/issues/1218
- https://github.com/beeware/briefcase/issues/1201
- https://github.com/beeware/briefcase/issues/1221
- https://github.com/beeware/briefcase/issues/1291 — specifically the bit about custom Info.plist keys; right now both of my apps need at least LSUIElement, but likely will need more

**Comment by glyph** [comment]
maybe also need codesigning-without-building-using-briefcase documentation, but I gather this is already at least possible and maybe the docs even exist, need to read up on that before filing anything there

**Comment by freakboy3742** [comment]
FYI: beeware/briefcase#2110 (currently in review) resolves the last of your blockers. If there's anything else standing in the way of Briefcase adoption, let us know.

**Comment by glyph** [comment]
At this point I think this is going to start coming down to the features of py2app, rather than encrust itself. Is there a tutorial for anywhere to use briefcase *just* for codesigning and archiving, on an existing app bundle? It would be easier to adopt it a bit at a time.

**Comment by glyph** [comment]
Oh wait I guess I can't quite just use it for codesigning, because I would need some way to invoke the code in https://github.com/beeware/briefcase/issues/1217 too, hmm.

**Comment by freakboy3742** [comment]
Yeah - you could probably trick Briefcase into thinking it had created the app, and so it could then sign it - but I wouldn't give any guarantees that any trick would be reliable in the long term - what you're describing definitely isn't an officially supported use case.

**Comment by glyph** [comment]
@freakboy3742 Trying to describe the major things that I think I'm still getting out of py2app.

**Comment by freakboy3742** [comment]
@glyph FYI- we've just added https://github.com/beeware/briefcase/issues/2326, so you can now use Briefcase to sign and notarize a `.app` produced by py2app (or another tool).

**Comment by glyph** [comment]
> FYI- we've just added [beeware/briefcase#2326](https://github.com/beeware/briefcase/issues/2326), so you can now use Briefcase to sign and notarize a `.app` produced by py2app (or another tool).

Awesome, thank you!

**Comment by glyph** [comment]
@freakboy3742 one thing that occurs to me here is that `py2app` runs `ibtool` on my `.xib` files for me. I don't really *need* this, but I do have to think about where it goes with respect to the rest of the build toolchain. Is this a thing that should live inside `briefcase`? Do you support plugins or something? I am reluctant to file an issue because it feels out of scope

**Comment by freakboy3742** [comment]
It sounds like it *could* be in scope. It would obviously be macOS specific, but I don't have any problem adding a macOS-specific part of the build process if the thing being added (a) is a common platform requirement, and (b) there's a meaningful, not-too-leaky abstraction that makes sense (e.g., "list of XIB files that need to be compiled", or "mapping of input XIB files to output file names"). XIB support definitely meets the requirements of (a); I haven't done enough work with standalone XIB files to know if there are other options that would need to be supported for (b). 

As for plugins - yes, we very much support the idea of having plugins. Internally, most of the core functionality of Briefcase is built as plugins; the existence of macOS platform support is a "platform plugin" that just happens to be installed by default. Generally speaking, I would prefer to add a plugin interface that allows adding arbitrary features over trying to bake all possible functionality into Briefcase.

There isn't a plugin hook for pre/post build processing right now, but that would be an obvious hook to add. An analogous request was made as beeware/briefcase#436, to support PySide/Qt "compile resources" invocation. beeware/briefcase#1865 could potentially be implemented using a similar interface. If you squint enough, beeware/briefcase#2073 could even be implemented as a "post-build" hook - but that might make more sense to keep internally given that preferring to include, not include, or *only* include PYC content are mutually exclusive and exhaustive options.

If we do add built-in support for XIB file compilation, it would be preferable to implement it using a plugin interface so we can squash both feature requests at once.

**Comment by freakboy3742** [comment]
I've just coalesced those feature requests into a unified ticket (#2933).
